loopback/loopback.go

```diff
@@ -0,0 +1,539 @@
+	name := filepath.Join(n.realPath, req.Name)
+	err := os.Mkdir(name, req.Mode)
+	if err != nil {
+		return nil, err
+	}
+	return &Node{realPath: name, isDir: true, fs: n.fs}, nil
+}
+
+var _ fs.NodeRemover = (*Node)(nil)
+
+// Remove implements fs.NodeRemover interface for *Node
+func (n *Node) Remove(ctx context.Context, req *fuse.RemoveRequest) (err error) {
+	name := filepath.Join(n.realPath, req.Name)
+	defer func() { log.Printf("%s.Remove(%s): error=%v", n.realPath, name, err) }()
+	return os.Remove(filepath.Join(n.realPath, req.Name))
+}
+
+var _ fs.NodeFsyncer = (*Node)(nil)
+
+// Fsync implements fs.NodeFsyncer interface for *Node
+func (n *Node) Fsync(ctx context.Context, req *fuse.FsyncRequest) (err error) {
+	defer func() { log.Printf("%s.Fsync(): error=%v", n.realPath, err) }()
+	return n.fs.getHandle(req.Handle).f.Sync()
+}
+
+func tToTv(t time.Time) (tv syscall.Timeval) {
+	tv.Sec = int64(t.Unix())
+	tv.Usec = int32(t.UnixNano() % time.Second.Nanoseconds() /
+		time.Microsecond.Nanoseconds())
+	return tv
+}
+
+var _ fs.NodeSetattrer = (*Node)(nil)
+
+// Setattr implements fs.NodeSetattrer interface for *Node
+func (n *Node) Setattr(ctx context.Context,
+	req *fuse.SetattrRequest, resp *fuse.SetattrResponse) (err error) {
+	defer func() { log.Printf("%s.Setattr(): error=%v", n.realPath, err) }()
+	if req.Valid.Size() {
+		if err = syscall.Truncate(n.realPath, int64(req.Size)); err != nil {
+			return nil
+		}
+	}
+
+	if req.Valid.Mtime() {
+		var tvs [2]syscall.Timeval
+		if !req.Valid.Atime() {
+			tvs[0] = tToTv(time.Now())
+		} else {
+			tvs[0] = tToTv(req.Atime)
+		}
+		tvs[1] = tToTv(req.Mtime)
+	}
+
+	if req.Valid.Handle() {
+		// panic("no idea what this is for")
+	}
+
+	if req.Valid.Mode() {
+		if err = os.Chmod(n.realPath, req.Mode); err != nil {
+			return err
+		}
+	}
+
+	if req.Valid.Uid() || req.Valid.Gid() {
+		if req.Valid.Uid() && req.Valid.Gid() {
+			if err = os.Chown(n.realPath, int(req.Uid), int(req.Gid)); err != nil {
+				return err
+			}
+		}
+		fi, err := os.Stat(n.realPath)
+		if err != nil {
+			return err
+		}
+		s := fi.Sys().(*syscall.Stat_t)
+		if req.Valid.Uid() {
+			if err = os.Chown(n.realPath, int(req.Uid), int(s.Gid)); err != nil {
+				return err
+			}
+		} else {
+			if err = os.Chown(n.realPath, int(s.Uid), int(req.Gid)); err != nil {
+				return err
+			}
+		}
+	}
+
+	if req.Valid.Flags() {
+		if err = syscall.Chflags(n.realPath, int(req.Flags)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+var _ fs.NodeRenamer = (*Node)(nil)
+
+// Rename implements fs.NodeRenamer interface for *Node
+func (n *Node) Rename(ctx context.Context,
+	req *fuse.RenameRequest, newDir fs.Node) (err error) {
+	np := filepath.Join(newDir.(*Node).realPath, req.NewName)
+	op := filepath.Join(n.realPath, req.OldName)
+	defer func() {
+		log.Printf("%s.Rename(%s->%s): error=%v",
+			n.realPath, op, np, err)
+	}()
+	return os.Rename(op, np)
+}
+
+// Handle represent an open file or directory
```

Comment:
Can there really be a `Handle` for a directory?  What would the Read/Write calls do?